snafu-derive/src/lib.rs

```diff
@@ -262,15 +284,46 @@ fn parse_snafu_enum(
                 }
             }
 
+            // Reverse the lists of fields where we may find duplicates so that we can call the
+            // second (and later) items the "duplicates" without uglier indexing.
+            source_fields.reverse();
+            backtrace_fields.reverse();
+            backtrace_delegates.reverse();
+
             let source_field = source_fields.pop();
-            // Report a warning if there are multiple?
+            for extra_field in source_fields.into_iter() {
+                errors.push(syn::Error::new(
+                    extra_field.name.span(),
+                    "Only one field can be designated as 'source'; found a duplicate here:",
+                ));
+            }
 
             let backtrace_field = backtrace_fields.pop();
-            // Report a warning if there are multiple?
+            for extra_field in backtrace_fields.into_iter() {
+                errors.push(syn::Error::new(
+                    extra_field.name.span(),
+                    "Only one field can be designated as 'backtrace'; found a duplicate here:",
+                ));
+            }
 
             let backtrace_delegate = backtrace_delegates.pop();
-            // Report a warning if there are multiple?
-            // Report a warning if delegating and our own?
+            for extra_field in backtrace_delegates.into_iter() {
+                errors.push(syn::Error::new(
+                    extra_field.name.span(),
+                    "Only one field can be designated as 'backtrace(delegate)'; found a duplicate here:",
+                ));
+            }
+
+            if backtrace_field.is_some() && backtrace_delegate.is_some() {
```

Comment:
Can this be something like `if let Some(backtrace_field,  backtrace_delegate) = (backtrace_field.as_ref(), backtrace_delegate.as_ref())` to avoid the `unwrap`s?

Reply:
Ah, I didn't know that syntax would work for combining `Option`s - I'll give it a shot.